NAPS2.Lib.Tests/Dependencies/DownloadControllerTests.cs

```diff
@@ -0,0 +1,189 @@
+﻿using NAPS2.Dependencies;
+using NAPS2.Sdk.Tests;
+using RichardSzalay.MockHttp;
+using Xunit;
+
+namespace NAPS2.Lib.Tests.Dependencies;
+
+public class DownloadControllerTests : ContextualTests
+{
+    private const string AnimalsSHA1 = "1739ff74f5d26f6c42d7a3fc438d615a93d38bd4";
+    private const string StockDogJpegSHA1 = "ca1f89964bdb345e97804784eefa49a00dbf7207";
+
+    private class EventHandlerWrapper
```

Comment:
A better way to verify events being invoked that doesn't require a helper class is to use an EventHandler mock. [Example](https://github.com/cyanfish/naps2/blob/e2523aefc5339da99506275670060b4ce3075aa2/NAPS2.Lib.Tests/Config/TransactionConfigScopeTests.cs#L63)